protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/FileBuilder.kt

```diff
@@ -131,7 +136,12 @@ internal open class FileBuilder(val namer: Namer = Namer.Standard, val supportMa
     ): File.Field.OneOf {
         val fields = oneofFields.map {
             // wrapper fields are not supposed to be used inside of oneof's
-            numberedFieldFromProto(ctx, it, mutableSetOf(), true) as File.Field.Numbered.Standard
+            numberedFieldFromProto(
+                ctx,
+                it,
+                mutableSetOf(),
+                ctx.fileDesc.syntax.let { it == "proto2" || it == null}
```

Comment:
I'm confused by the purpose of this change. I guess it works correctly if the tests are all passing, but it doesn't make sense to me. Previously we would pass in `true` here for the `alwaysRequired` parameter of `numberedFieldFromProto()` because a field that is part of a `oneof` does not have its own presence -- the presence is tracked via the enclosing `oneof` field.

Now you're only passing in `true` if the `oneof` is in a proto2 file but not in a proto3 file. However I don't think there are any differences in how `oneof`s are treated in proto2 vs proto3. So it seems odd to make that distinction here. What is the impact of leaving this as always `true`? and is there somewhere else in the code that we can account for this impact instead of doing it here?

Reply:
I can't recall. When I analyse the code now, the change seems unnecessary and without any impact... Maybe I should revert it.
Could you explain semantic of `alwaysRequired` argument and why is it true for all one of's? This might have gotten me confused and as I was dealing with proto3 presence only I might have chosen to change it only for proto3.

Reply:
`alwaysRequired` is probably poorly named. It is used for fields that should not be stored as a nullable type in Kotlin even if they normally would be nullable based on the proto field declaration. For example, a field with `Type.INT32` and `FieldDescriptorProto.Label.OPTIONAL` (a combination that is valid in proto2) has "presence" and would normally be stored as a nullable variable in Kotlin. However, if that field is inside of a `oneof` then it should not be stored as a nullable variable because its presence/absence is represented by the `sealed class` that contains it.

Hopefully that explained it better? But honestly the usage of the `alwaysRequired` parameter is quite confusing when I look at how it's used in `CodeGenerator`. I think after your PR merges I'll try to refactor that code to get rid of this concept of "always required" that is specific to oneofs and instead base everything on the "presence" concept.